db/migrate/20170425154145_add_sti_to_middlewares.rb

```diff
@@ -0,0 +1,67 @@
+class AddStiToMiddlewares < ActiveRecord::Migration[5.0]
+  class MiddlewareDatasource < ApplicationRecord
+    self.table_name = 'middleware_datasources'
+  end
+
+  class MiddlewareDeployment < ApplicationRecord
+    self.table_name = 'middleware_deployments'
+  end
+
+  class MiddlewareDomain < ApplicationRecord
+    self.table_name = 'middleware_domains'
+  end
+
+  class MiddlewareMessaging < ApplicationRecord
+    self.table_name = 'middleware_messagings'
+  end
+
+  class MiddlewareServer < ApplicationRecord
+    self.table_name = 'middleware_servers'
+  end
+
+  class MiddlewareServerGroup < ApplicationRecord
+    self.table_name = 'middleware_server_groups'
+  end
+
+  def up
+    add_column :middleware_datasources, :type, :string, :null => false, :default => ''
```

Comment:
see https://github.com/ManageIQ/manageiq/pull/13965 for discussions around `:type` and indeces.

I think it should be just 
```
add_column :cloud_object_store_containers, :type, :string
add_index :cloud_object_store_containers, :type
```